agent/hcp/telemetry/otel_sink.go

```diff
@@ -77,14 +80,31 @@ func NewOTELSink(opts *OTELSinkOpts) (*OTELSink, error) {
 		return nil, fmt.Errorf("ferror: provide valid context")
 	}
 
+	logger := hclog.FromContext(opts.Ctx).Named("otel_sink")
+
 	// Setup OTEL Metrics SDK to aggregate, convert and export metrics.
-	res := resource.NewSchemaless()
+
+	filterList, err := NewFilterList(opts.Filters)
+	if err != nil {
+		logger.Error("Failed to initialize all filters: %w", err)
+	}
+
+	attrs := make([]attribute.KeyValue, len(opts.Labels))
+	for k, v := range opts.Labels {
```

Comment:
I think this line is initializing `attrs` with empty `attribute.KeyValue`s. Eg: https://go.dev/play/p/Z-jQo-KLVVw

Reply:
oh good catch! thank you 🙏🏽 I did not expect that to be the behaviour :O